### PR TITLE
add sent tx to transactions

### PR DESCRIPTION
**Pull request by coinmaverick**

### CODE REVIEW

File: app/utils/parseMemo.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import hex from 'hex-string';
+
+const parseMemo = (memoHex: string): string | null => {
+  if (!memoHex || memoHex.length < 2) return null;
+
+  // First, check if this is a memo (first byte is less than 'f6' (246))
+  if (parseInt(memoHex.substr(0, 2), 16) >= 246) return null;
+
+  // Else, parse as Hex string
+  const textDecoder = new TextDecoder();
```
Comment (j4ys0n) [inline app/utils/parseMemo.js:10]: nice - i didn't know about `TextDecoder`. looks pretty handy. https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.textdecoder.html

File: app/utils/SentTxStore.js [hunk 2]
```diff
@@ -3,8 +3,26 @@ import path from 'path';
...
+      await fs.promises.writeFile(fileName, JSON.stringify([sentTx]));
+      return;
+    }
+
+    fs.readFile(fileName, async (err, data) => {
+      const json = JSON.parse(data);
+
+      json.push(sentTx);
+
+      fs.promises.writeFile(fileName, JSON.stringify(json));
```
Comment (j4ys0n) [inline app/utils/SentTxStore.js]: add `await`

File: app/utils/SentTxStore.js [hunk 3]
```diff
@@ -3,8 +3,26 @@ import path from 'path';
...
 export default class SentTxStore {
+  static async writeSentTx(sentTx) {
+    const fileName = await this.locateSentTxStore();
+
+    if (!fs.existsSync(fileName)) {
+      await fs.promises.writeFile(fileName, JSON.stringify([sentTx]));
+      return;
+    }
+
+    fs.readFile(fileName, async (err, data) => {
```
Comment (j4ys0n) [inline app/utils/SentTxStore.js:17]: should this be `await fs.promises.readFile(...`?

File: app/utils/SentTxStore.js [hunk 4]
```diff
@@ -26,24 +44,29 @@ export default class SentTxStore {
...
-        transction.detailedTxns[0].memo = s.memo;
+        transction.detailedTxns[0].address = amounts[0].address;
+        transction.detailedTxns[0].amount = amounts[0].amount;
+        transction.detailedTxns[0].memo = amounts[0].memo ? parseMemo(amounts[0].memo) : '';
 
         return transction;
       });
     } catch (err) {
       // If error for whatever reason (most likely, file not found), just return an empty array
+      console.error(err);
```
Comment (j4ys0n) [inline app/utils/SentTxStore.js:69]: good call

**Review by j4ys0n — COMMENTED** [review]
looks good! just a couple comments

File: app/utils/SentTxStore.js (hunk 3 above)
Comment (coinmaverick) [reply app/utils/SentTxStore.js:17]: no, don't think so. we're not waiting for `fs.readFile` itself. we invoke the function, which is an async function, it gets added to the microtask queue to be executed when the call stack is empty, and once it runs then our async callback is invoked. we're not dependant on it anywhere else so we can just queue it up 👍

**Review by j4ys0n — APPROVED** [review]

**Pull request merged**